src/toxcore/dht_new/codec.rs

```diff
@@ -0,0 +1,86 @@
+/*
+    Copyright (C) 2013 Tox project All Rights Reserved.
+    Copyright © 2016-2017 Zetok Zalbavar <zexavexxe@gmail.com>
+    Copyright © 2018 Namsoo CHO <nscho66@gmail.com>
+
+    This file is part of Tox.
+
+    Tox is libre software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Tox is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Tox.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/*! Codec for encoding/decoding DHT Packets & DHT Request packets using tokio-io
+*/
+
+use toxcore::dht_new::packet::*;
+use toxcore::dht_new::binary_io::*;
+
+use std::io;
+use tokio_core::net::UdpCodec;
+use std::net::SocketAddr;
+use std::io::{Error, ErrorKind};
+
+/// Type representing Dht UDP packets.
+pub type DhtUdpPacket = (SocketAddr, DhtBase);
+
+/// Type representing received Dht UDP packets.
+pub type DhtRecvUdpPacket = (SocketAddr, Option<DhtBase>);
+
+/**
+SendNodes
```

Comment:
I think it should be explained better why we use SendNodes here to estimate maximum size of DhtPacket.

Reply:
Good catch!